src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts

```diff
@@ -0,0 +1,480 @@
+                    .execute()
+                    .done((accessResult: Raven.Client.Documents.Operations.Replication.ReplicationHubAccessResult) => {
+                        self.processResults(accessResult);
+                        this.editedReplicationAccessItem(null);
+                    });
+            })
+            .always(() => this.spinners.saveReplicationAccess(false));
+    } 
+   
+    cancelHubTaskOperation() {
+        this.goToOngoingTasksView();
+    }
+    
+    cancelReplicationAccessOperation() {
+        this.editedReplicationAccessItem(null);
+        this.spinners.generateCertificate(false);
+    }  
+
+    addNewReplicationAccess() {
+        if (this.editedReplicationAccessItem() && this.editedReplicationAccessItem().dirtyFlag().isDirty()) {
+            this.warnAboutUnsavedChanges()
+                .done(result => {
+                    if (result.can) {
+                        this.addItem();
+                    }
+                });
+        } else {
+            this.addItem();
+        }
+    }
+    
+    addItem() {
+        this.editedReplicationAccessItem(replicationAccessHubModel.empty(this.editedHubTask().withFiltering()));
+        this.initTooltips();
+    }
+   
+    editReplicationAccessItem(replicationAcessItem: replicationAccessHubModel) {
+        if (this.editedReplicationAccessItem() && this.editedReplicationAccessItem().dirtyFlag().isDirty()) {
+            this.warnAboutUnsavedChanges()
+                .done(result => {
+                    if (result.can) {
+                        this.editItem(replicationAcessItem);
+                    }
+                });
+        } else {
+            this.editItem(replicationAcessItem);
+        }
+    }
+   
+    editItem(replicationAcessItem: replicationAccessHubModel) {
+        // work on a copy, not on original
+        let copyOfAccessItem = replicationAccessHubModel.clone(replicationAcessItem);
+        this.editedReplicationAccessItem(copyOfAccessItem);
+
+        this.initTooltips();
+    }
+    
+    cloneReplicationAccessItem() {
+        if (this.editedReplicationAccessItem().dirtyFlag().isDirty()) {
+            this.warnAboutUnsavedChanges()
+                .done(result => {
+                    if (result.can) {
+                        this.cloneItem();
+                    }
+                });
+        } else {
+            this.cloneItem();
+        }
+    }
+
+    cloneItem() {
+        const editedItem = this.editedReplicationAccessItem();
+        let cloneItem = new replicationAccessHubModel("", null, editedItem.hubToSinkPrefixes(), editedItem.sinkToHubPrefixes(), editedItem.filteringPathsRequired());
+        this.editedReplicationAccessItem(cloneItem);
+        this.initTooltips();
+    }
+
+    warnAboutUnsavedChanges() {
+        return this.confirmationMessage("Unsaved changes",
+            `<div>You have unsaved changes. How do you want to proceed?</div>`, {
```

Comment:
do we need this div here + html: html param?

Reply:
done